relay-server/src/actors/events.rs

```diff
@@ -276,6 +276,29 @@ impl EventProcessor {
         Self { config }
     }
 
+    /// Validates any and all measurements in the envelope.
+    fn process_measurements(
+        &self,
+        state: &mut ProcessEnvelopeState,
+    ) -> Result<(), ProcessingError> {
+        log::trace!("processing measurements");
+
+        let is_transaction = state.event_type() == Some(EventType::Transaction);
+
+        let event = match state.event.value_mut() {
+            Some(event) => event,
+            None => return Ok(()),
+        };
+
+        if !is_transaction {
+            // Only transaction events may have a measurements interface
+            event.measurements = Annotated::empty();
+            return Ok(());
+        }
+
+        Ok(())
+    }
```

Comment:
This is something we should put into the [StoreNormalizer](https://github.com/getsentry/relay/blob/569678659bac656d6c4a6edcbdf3381677b5eaa2/relay-general/src/store/normalize.rs#L83) since this mutates the event payload.

Reply:
I've moved the logic to the StoreNormalizer and updated the tests as appropriate.